File: typescript/mergeArray.ts
const firstArray = [0, 3, 4, 22, 31];
const secondArray = [4, 6, 13, 30];

const mergeArray = (firstArray: number[], secondArray: number[]): number[] => {
	if (!Array.isArray(firstArray) || !Array.isArray(secondArray)) {
		throw new Error('Parameters must be arrays');
	}
	const result: number[] = [];
	const finalLength = firstArray.length + secondArray.length - 2;
	let f = 0;
	let s = 0;

	for (let i = 0; i <= finalLength; i++) {
		if (f >= firstArray.length) {
			result.push(...secondArray.slice(s));
			break;
		} else if (s >= secondArray.length) {
			result.push(...firstArray.slice(f));
			break;
		} else if (firstArray[f] < secondArray[s]) {
			result.push(firstArray[f]);
			f++;
		} else {
			result.push(secondArray[s]);
			s++;
		}
	}

	return result;
};

console.log(mergeArray(firstArray, secondArray));

File: python/merge_array.py
f_array = [0, 3, 4, 22, 31]
s_array = [4, 6, 13, 30]


def merge_array(first_array: list[int], second_array: list[int]):
    if not isinstance(first_array, list) or not isinstance(second_array, list):
        raise ValueError("the arguments must be lists")

    result: list[int] = []
    final_length = len(first_array) + len(second_array) - 2
    f_index = 0
    s_index = 0

    for i in range(0, final_length):
        if f_index >= len(first_array):
            result.extend(second_array[s_index:])
            break
        elif s_index >= len(second_array):
            result.extend(first_array[f_index:])
            break
        elif first_array[f_index] < second_array[s_index]:
            result.append(first_array[f_index])
            f_index += 1
        else:
            result.append(second_array[s_index])
            s_index += 1

    return result


print(merge_array(f_array, s_array))
